### DISCUSSION TITLE
UKeepler, the new Udeler coming soon

### DISCUSSION THREAD

**Discussion by heliomarpm** [heliomarpm/udemy-downloader-gui#125] [Announcements]
# Udeler, soon to be called UKeepler

Hello everyone how are you?

It's been almost 3 years since I took over this Udeler project, since then I've been putting a lot of effort into improving the coding, architecture and updating third-party libraries.

Still, many already solved problems come back to haunt us, new features are difficult to implement and change.

So I started creating a new application, using current libraries and frameworks, with segregated code and improved design.

I hope that with this version I can attract other developers, do everything myself in my spare time, it's exhausting lol.

Below is a prototype of the new UKeepler application.

Some features I intend to keep and add:
* Language applied without restart
* Downloads in 1080p
* Translation of subtitles into the chosen language when not available
* Settings saved automatically, no need to click a button
* Course information cached locally
* Identify when the downloaded course received an update
* Allows you to select which file to download.
* And most importantly, you'll know if there's an encrypted class before starting the download.
* Among other things

Please leave your comment here about what you expect from this new tool and take the opportunity to include a feature request.

## Remembering that this new application will follow the same guidelines, so we will continue with the policy of not breaking the encryption of videos protected by copyright.

![Prototipo_UKeepler](https://user-images.githubusercontent.com/13087389/236926293-4b3cd359-d30c-4d2e-9055-14cb2ba4894c.gif)

**Comment by San-blip55** [comment]
Its been unable to download DRM protected videos. Mind checking it?

**Reply by heliomarpm** [reply]
This isn't possible.

Try this one https://github.com/Puyodead1/udemy-downloader

**Reply by FAUSTOCG** [reply]
Hi, I know that DRM videos can not be downloaded but subtitles and extra files used to have no problem before, I have noticed that now subtitles and extra files are not downloading  does the new realease fix this issue?
thank you for your hard work
best regards

**Reply by heliomarpm** [reply]
> Hi, I know that DRM videos can not be downloaded but subtitles and extra files used to have no problem before, I have noticed that now subtitles and extra files are not downloading  does the new realease fix this issue?
> thank you for your hard work
> best regards

I have no reports of this problem.  Please open an issue and provide as much information as possible so I can reproduce.  But before that you can try to go into settings, save and try again

**Reply by MUSTAFAALI06** [reply]
Teacher, first of all, good luck. This is the new udeler version 1.12.7. I think there is a problem? It doesn't download the videos properly. Best regards and good luck.


Hocam kolay gelsin öncelikle. Bu yeni çıkarmış olduğunuz udeler. 1.12.7 Versiyonu. Sorun var galiba? Bir türlü videoları doğru dürüst indirmiyor. Saygılarımla kolay gelsin.

**Comment by ntk-quanxi** [comment]
Sir, when you released this new version ? I cant not download enrolled course in my business account with udeler.

**Reply by heliomarpm** [reply]
Sorry UKeepler is still in development stage.
What kind of errors are you having in downloading using Udeler?

If it's not due to DRM protection, you can open an issue with as much detail as possible and I'll check as soon as possible

**Comment by MUSTAFAALI06** [comment]
Teacher, first of all, good luck. This is the new oudeler you just released. Version 1.12.7. I think there's a problem? It won't download the videos properly. Best regards and good luck.

**Comment by MUSTAFAALI06** [comment]
It cuts videos in half. And it does not exceed the extra files it downloads. I am using Windows 11. What could be the cause of this, sir?

**Comment by heliomarpm** [comment]
> It cuts videos in half. And it does not exceed the extra files it downloads. I am using Windows 11. What could be the cause of this, sir?

I haven't found any issues related to this.  Open an issue with more details and I'll take a look

**Comment by TechnicalExpo** [comment]
Hello Bro! You are doing great, Udeler has some features like Authenticator and Token Access but they are not working.... make it able to work... we provide access token key but it's not working

Second in the details you gave the extension file of udeler when we click it give 404 error file not found.

**Reply by heliomarpm** [reply]
Hello my friend, I hope you are well.  If you are using a business account token, don't forget to also provide the subdomain name.  As for the extension, it is not my creation, the most I can offer you at this moment is a document explaining how to obtain the token.

https://heliomarpm.notion.site/heliomarpm/How-to-get-an-Access-Token-b0ce5f89073c4965b47721197dcb2497

**Reply by TechnicalExpo** [reply]
Hello! I'm Fine hope you will be fine too. 
Bro I have am using Udemy Subscription Plan account, I know how to get Token, but the problem is when I copy and past the token in the Aceess token in the Udeler it's not opening, it just logged in through Credentials, but I need to give someone the access token so he can download the course but I don't want to share cred.

I hope you understand what I want to say...

Inshort is it possible to Open or download course through just access token....?

**Reply by heliomarpm** [reply]
Yes, it is possible to connect using just the token and download all purchased courses with the exception of encrypted courses.
Please open an issue detailing the problem so I can investigate.

**Comment by DoanPham37** [comment]
> Có, có thể kết nối chỉ bằng mã thông báo và tải xuống tất cả các khóa học đã mua ngoại trừ các khóa học được mã hóa. Vui lòng mở một vấn đề nêu chi tiết vấn đề để tôi có thể điều tra.

I have kid & key, how do I use it on udeler

**Comment by Canmemis** [comment]
Hello,
When will UKeepler be released?
Currently, when I want to download from Udeler, it downloads as 720p even though the video quality is 1080p.

**Reply by heliomarpm** [reply]
This is an issue with older versions. Which version are you using?
Also make sure it is set to download at the highest quality.

**Reply by Canmemis** [reply]
https://github.com/heliomarpm/udemy-downloader-gui/releases
I am using the 64 bit version available here.
in the settings I selected the option to download in highest resolution. but it downloads in 720p.
![image](https://github.com/heliomarpm/udemy-downloader-gui/assets/88040387/4e685631-d767-4c23-b444-0cc71ae28b79)
![image](https://github.com/heliomarpm/udemy-downloader-gui/assets/88040387/8d312453-dada-4255-987d-cc3b41c62941)

training normally supports 1080p.
![image](https://github.com/heliomarpm/udemy-downloader-gui/assets/88040387/304a3036-f458-405e-aa0c-d372b17e253b)

**Reply by heliomarpm** [reply]
Please open an issue, and provide the name of the course so I can investigate

**Reply by Canmemis** [reply]
I created the topic.

**Comment by SakAwans** [comment]
I'm not here to complain about a bug or issue; I'm just here to appreciate your work, brother <3. I don't have words to describe how grateful I am for this program because it has helped me a lot. Please keep updating this program, and maybe someday, if I get a job, I will definitely donate some bucks for you ;). Again, much love and respect, brother.

**Reply by heliomarpm** [reply]
Yes, I do what I do with pleasure, it is gratifying to be able to help many.

It's been a while since I posted an update, but the work continues.

**Comment by FAUSTOCG** [comment]
Hello my friend first I want to thank you for this great app udeler, but wanted to let you know that there is a glitch i have downloaded a full course that just has been released but when playing the videos the audios have glitches but when playing them directly from the web audio has no glitches I download it again in different computers with the same audio glitches bough another course and it happened again videos play great buy audios have glitches or could be some sort of DRM implementation by Udmy?

thanks for keeping this great alive app have a great week
